src/gui/qgsabstractdatasourcewidget.h

```diff
@@ -60,6 +54,21 @@ class GUI_EXPORT QgsAbstractDataSourceWidget : public QDialog
      */
     void setBrowserModel( QgsBrowserModel *model );
 
+    /**
+     * Returns the dialog map canvas
+     * \see setMapCanvas()
+     *
+     * \since QGIS 3.26
+     */
+    virtual QgsMapCanvas *mapCanvas() = 0;
```

Comment:
No need to make this one pure virtual:

```suggestion
    virtual QgsMapCanvas *mapCanvas() { return mMapCanvas; }
```